src/commands/importIdentifier.ts

```diff
@@ -0,0 +1,108 @@
+import * as vscode from 'vscode';
+import { CommandNames } from './constants';
+import * as request from 'request-promise-native';
+import * as cheerio from 'cheerio';
+import * as LRU from 'lru-cache';
+import * as _ from 'lodash';
+import * as yaml from 'js-yaml';
+
+const askHoogle = async (variable: string): Promise<any> => {
+  return await request({
+    url: `https://hoogle.haskell.org/?hoogle=${variable}&scope=set%3Astackage&mode=json`,
+    json: true
+  }).promise();
+};
+
+const withCache = <T, U>(cache: LRU.Cache<T, U>, f: (a: T) => U) => (a: T) => {
+  const maybeB = cache.get(a);
+  if (maybeB) {
+    return maybeB;
+  } else {
+    const b = f(a);
+    cache.set(a, b);
+    return b;
+  }
+};
+
+const cache: LRU.Cache<string, Promise<any>> = LRU({
+  // 1 MB
+  max: 1000 * 1000,
+  length: (r: any) => JSON.stringify(r).length
+});
+
+const askHoogleCached = withCache(cache, askHoogle);
+
+const doImport = async (arg: { mod: string; package: string }): Promise<void> => {
+  const editor = vscode.window.activeTextEditor;
+  if (!editor) {
+    return;
+  }
+
+  const document = editor.document;
+  let edit = new vscode.WorkspaceEdit();
+
+  const lines = document.getText().split('\n');
+  const moduleLine = lines.findIndex(line => line.startsWith('module'));
+  const revInputLine = lines.reverse().findIndex(l => l.startsWith('import'));
+  let nextInputLine = revInputLine !== -1 ? lines.length - 1 - revInputLine : moduleLine === -1 ? 0 : moduleLine + 1;
+
+  if (!lines.some(line => new RegExp('^import.*' + _.escapeRegExp(arg.mod)).test(line))) {
+    edit.insert(document.uri, new vscode.Position(nextInputLine, 0), 'import ' + arg.mod + '\n');
+  }
+
+  const hpackDoc = await vscode.workspace.openTextDocument(vscode.workspace.rootPath + '/package.yaml');
```

Comment:
What if `package.yaml` does not exist? In many setups even with `stack` people not using `hpack`, not mentioning `cabal-install` users.

Reply:
Perhaps this part can be enabled via a config setting?

Reply:
It will probably require a lot more effort to support Cabal files (as opposed to hpack, which is YAML and easily parsed) without resorting to [brittle regex hacks](https://github.com/soates/Auto-Import/compare/master...chrismwendt:master#diff-c1342d7f94e21b14d38de827d9a2a0daR62) since there doesn't appear to be a Cabal file parser in JavaScript. That said, a missing `package.yaml` should be handled gracefully.

Reply:
I was thinking more on the lines of enabling it via the approach you have now, or disable it completely.

Programmatically changing cabal files is not something that works reliably at the moment.

Reply:
It could at least detect if `package.yaml` is present or not, and if not then do not even provide  this command...
I would prefer this behaviour to config options.

Also a side question: when `package.yaml` is changed and new dependencies are added, HIE smart enough to pick these changes up, install dependencies, etc.? Otherwise there is a risk that the IDE will show errors until user manually calls `hpack`/`stack` and re-builds the project...

Reply:
I'll avoid modifying `package.yaml` if it doesn't exist.

After adding a new dependency, I always have to reload the VS Code window to get hie to recognize the new modules, but that's another issue entirely.